test/e2e/util_gatewayapi_test.go

```diff
@@ -103,19 +103,51 @@ func assertCrdExists(t *testing.T, crdname string) (string, error) {
 	return crdVersion, err
 }
 
+// deleteExistingCRDs deletes if the CRD of the given name exists and returns an error if not.
+func deleteExistingCRDs(t *testing.T, crdname string) error {
+	t.Helper()
+	crd := &apiextensionsv1.CustomResourceDefinition{}
+	name := types.NamespacedName{Namespace: "", Name: crdname}
+
+	err := wait.PollUntilContextTimeout(context.Background(), 1*time.Second, 30*time.Second, false, func(context context.Context) (bool, error) {
+		if err := kclient.Get(context, name, crd); err != nil {
+			t.Logf("failed to get crd %s: %v", name, err)
+			return false, nil
+		}
+		crdConditions := crd.Status.Conditions
+		for _, c := range crdConditions {
+			if c.Type == apiextensionsv1.Established && c.Status == apiextensionsv1.ConditionTrue {
+				return true, nil
+			}
+		}
+		t.Logf("failed to find crd %s to be Established", name)
+		return false, nil
+	})
+	if err != nil {
+		t.Errorf("failed to get crd %s: %v", name, err)
+		return err
+	}
+	err = kclient.Delete(context.Background(), crd)
+	if err != nil {
+		t.Errorf("failed to delete crd %s: %v", name, err)
+		return err
+	}
```

Comment:
I believe this will mark the CRD for deletion, not-block, and continue on. Deletion can take a couple of seconds. A potential issue is that the CRD is still available, but being deleted, and the `EnsureCRDs` function finds them without being delete, and passes because they still look good even though they are in the process of being deleted.

To be safe, I'd consider blocking here. One example of blocking or waiting for deletion is [deleteIngressController](https://github.com/openshift/cluster-ingress-operator/blob/dc4e7f9a511ae028db8149609a1f4b80f7b8a500/test/e2e/operator_test.go#L4263-L4272).

Reply:
@gcs278 I have added the poll because the IngressController has multiple operands, which means it takes some time to delete. However, CRDs are deleted quite quickly. When I receive my first poll CRD, it gives me a new CRD with a different UUID, and it ends up stuck in a loop searching for a new CRD.
```
// if new CRD got recreated while the poll ensures the CRD is deleted.
if newCRD != nil && newCRD.UID != crd.UID {
	return true, nil
}
```

Reply:
> When I receive my first poll CRD, it gives me a new CRD with a different UUID

Sorry, I might be confused, isn't this a good thing? Not sure why it'd get stuck in this loop.

Looks like the test passed, are you saying it's not working correctly still or did you figure it out?

Reply:
The problem here I am trying to address is that if I remove the code section for 
```
// if new CRD got recreated while the poll ensures the CRD is deleted.
if newCRD != nil && newCRD.UID != crd.UID {
	return true, nil
}
```
The problem will be the reconcile loop is creating the CRD again pretty quickly and when I am trying get the CRD in the poll section it is already been recreated, in that case the `if kerrors.IsNotFound(err)` condition will never satisfied and tests will end-up failing due to context deadline exceeded.
This is the reason Why I have validated it with old CRD's UID.

Reply:
Oh yes, that sounds very reasonable, better to be precise to avoid race conditions.